modules/publication/ajax/FileUpload.php

```diff
@@ -114,10 +116,9 @@ function uploadPublication()
         insertKeywords($pubID);
         // INSERT INTO publication_parameter_type_rel
         insertVOIs($pubID);
-        // INSERT INTO publication_users_edit_perm_rel
     } catch (Exception $e) {
         cleanup($pubID);
-        showError($e->getMessage());
+        showError($e->getMessage(), $e->getCode());
```

Comment:
I think this would be better as just `showError($e->getMessage())`. The way that this is written makes it look like you want to use the PHP Exception codes as HTTP Response Codes and these systems aren't the same. Since your default case is response code `500` this shouldn't affect functionality but will make the code a bit simpler to understand.

Reply:
if i got rid of this argument, wouldn't that just make every error a 500? i get that php exception & http response codes aren't necessarily equivalent / interchangeable, but so long as i'm using it consistently though out this module's code it should be ok, no? 
any exception that gets thrown from a library / external code to it would (most likely) default to 500 which should be good enough imo :thinking: 
let me know what you think, would definitely like to make the error handling here more robust but not exactly sure how to go about it lol ¯\_(ツ)_/¯

Reply:
It would make every error here a 500 yes but what I'm saying is that the PHP Exception errors will also always be a 500 error (as it falls through to your default switch-case statement). 

I like the usage elsewhere in the code I just think in this part specifically it's confusing but not a huge deal.